lib/net.js

```diff
@@ -133,6 +134,7 @@ function Socket(options) {
   this._handle = null;
   this._parent = null;
   this._host = null;
+  this._timeout = null;
```

Comment:
Rather than introducing yet another `_` prefixed property, I'd much prefer to use a Symbol.
Also, given that I'm following an identical pattern in the http2 work, I'd like that Symbol to be reusable (perhaps as an export off the `internal/timers.js` this PR creates